### PR TITLE
feat: quote item list shall be editable when quote is switched to edit mode [CXSPA-4231]

### PR DESCRIPTION
**Pull request by Uli-Tiger**
When quote is in edit mode, one can update the item quantity as well as remove any item from the quote cart.

### CODE REVIEW

File: feature-libs/quote/components/details/cart/quote-details-cart.component.ts [hunk 1]
```diff
@@ -4,28 +4,46 @@
...
 
+  ngOnInit(): void {
+    // if anything in the cart changes, reload the quote
+    this.subscription = this.eventService.get(CartEvent).subscribe(() => {
+      this.eventService.dispatch({}, QuoteDetailsReloadQueryEvent);
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscription?.unsubscribe();
```
Comment (ChristophHi) [inline feature-libs/quote/components/details/cart/quote-details-cart.component.ts]:
```suggestion
    this.subscription.unsubscribe();
```

File: feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts [hunk 2]
```diff
@@ -61,6 +61,7 @@ export class CartItemListComponent implements OnInit, OnDestroy {
   @Input() cartId: string;
 
   protected _items: OrderEntry[] = [];
+  protected _foreReRender: boolean = false;
```
Comment (ChristophHi) [inline feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts]:
```suggestion
  protected _forceReRender: boolean = false;
```

File: feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts [hunk 3]
```diff
@@ -119,6 +120,7 @@ export class CartItemListComponent implements OnInit, OnDestroy {
         this.cartId = context.cartId;
       }
       if (context.items !== undefined) {
+        this._foreReRender = true;
```
Comment (ChristophHi) [inline feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts]: That means we re-render _all_ entries whenever they are provided.
Shouldn't we set forceReRender only if readOnly changes?

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts (hunk 3 above)
Comment (Uli-Tiger) [reply feature-libs/cart/base/components/cart-shared/cart-item-list/cart-item-list.component.ts]: i think you are right - it's sufficient to do if the read-only flag changes

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**